extensions/ConfirmEdit/ReCaptcha.class.php

```diff
@@ -1,69 +1,60 @@
 <?php
 
 class ReCaptcha extends SimpleCaptcha {
-	// reCAPTHCA error code returned from recaptcha_check_answer
-	private $recaptcha_error = null;
+
+	const VERIFY_URL = 'https://www.google.com/recaptcha/api/siteverify';
+	const API_URL_TEMPLATE = 'https://www.google.com/recaptcha/api.js?hl=$1';
 
 	/**
 	 * Displays the reCAPTCHA widget.
-	 * If $this->recaptcha_error is set, it will display an error in the widget.
-	 *
 	 */
 	function getForm() {
-		global $wgReCaptchaPublicKey, $wgReCaptchaTheme;
-
-		$useHttps = ( isset( $_SERVER['HTTPS'] ) && $_SERVER['HTTPS'] == 'on' );
-		$js = 'var RecaptchaOptions = ' . Xml::encodeJsVar( array( 'theme' => $wgReCaptchaTheme /*Wikia change */ /*, 'tabindex' => 1 */ /*Wikia change end*/) );
+		$siteKey = F::app()->wg->ReCaptchaPublicKey;
+		$theme = SassUtil::isThemeDark() ? 'dark' : 'light';
 
-		return Html::inlineScript( $js ) . recaptcha_get_html( $wgReCaptchaPublicKey, $this->recaptcha_error, $useHttps );
+		return '<div class="g-recaptcha" data-sitekey="' . $siteKey . '" data-theme="' . $theme . '"></div>';
 	}
 
 	/**
-	 * Calls the library function recaptcha_check_answer to verify the users input.
-	 * Sets $this->recaptcha_error if the user is incorrect.
-	 * @return boolean
+	 * Calls the API method siteverify to verify the users input.
 	 *
+	 * @return boolean
 	 */
-	function passCaptcha() {
-		global $wgReCaptchaPrivateKey, $wgRequest;
+	public function passCaptcha() {
+		$verifyUrl = $this->getVerifyUrl();
 
-		// API is hardwired to return wpCaptchaId and wpCaptchaWord, so use that if the standard two are empty
-		$challenge = $wgRequest->getVal( 'recaptcha_challenge_field', $wgRequest->getVal( 'wpCaptchaId' ) );
-		$response = $wgRequest->getVal( 'recaptcha_response_field', $wgRequest->getVal( 'wpCaptchaWord' ) );
+		$responseObj = Http::get( $verifyUrl, 'default', [
+			'noProxy' => true,
+			'returnInstance' => true
+		] );
 
-		if ( $response === null ) {
-			// new captcha session
-			return false;
-		}
+		if ( $responseObj->getStatus() === 200 ) {
+			$response = json_decode( $responseObj->getContent() );
 
-		// Compat: WebRequest::getIP is only available since MW 1.19.
-		$ip = method_exists( $wgRequest, 'getIP' ) ? $wgRequest->getIP() : wfGetIP();
-
-		$recaptcha_response = recaptcha_check_answer(
-			$wgReCaptchaPrivateKey,
-			$ip,
-			$challenge,
-			$response
-		);
-
-		if ( !$recaptcha_response->is_valid ) {
-			$this->recaptcha_error = $recaptcha_response->error;
-			return false;
+			if ( $response->success === true ) {
+				return true;
+			}
 		}
```

Comment:
That's a good idea nAndy.  We're moving the capcha stuff into the extensions/wikia directory and cleaning it up, so I'll put that in then.